rfcs/20190729-keras-preprocessing.md

```diff
@@ -0,0 +1,738 @@
+
+```python
+@classmethod
+def from_config(self, config):
+    # Returns instance
+
+def get_config(self, config):
+    # Returns JSON-serializable config dict
+
+def preview(self, data, save_to_directory=None, save_prefix=None, save_format='png'):
+    """Enables users to preview the image augmentation configuration.
+
+    # Arguments
+        data: Image datra. Could be strings (a list of image paths), a list of PIL image instances,
+            a list of arrays, or a list of eager tensors.
+        save_to_directory: Directory to save transformed images. Mandatory if not in a notebook.
+            If in a notebook and this is not specified, images are displayed in-line.
+        save_prefix: String, filename prefix for saved images.
+        save_format: String, extension for saved images.
+    """
+
+def apply(self, data, seed=None):
+    # Takes an image or batch of images (PIL instance(s), array, tensor)
+    # and returns a Numpy array (or tensor) of standardized and augmented data.
+    # Note: at this time we have the methods `standardize` and `random_transform`;
+    # this new one does both at the same time.
+```
+
+#### Arrays methods
+
+```python
+# CURRENT
+
+def flow(
+    self,
+    x,
+    y=None,
+    batch_size=32,
+    shuffle=True,
+    sample_weight=None,
+    seed=None,
+    save_to_dir=None,
+    save_prefix='',
+    save_format='png',
+    subset=None):
+    """Returns Sequence instance (generator).
+
+    # Arguments
+        x: Input data. Numpy array of rank 4 or a tuple.
+            If tuple, the first element
+            should contain the images and the second element
+            another numpy array or a list of numpy arrays
+            that gets passed to the output
+            without any modifications.
+            Can be used to feed the model miscellaneous data
+            along with the images.
+            In case of grayscale data, the channels axis of the image array
+            should have value 1, in case
+            of RGB data, it should have value 3, and in case
+            of RGBA data, it should have value 4.
+        y: Labels.
+        batch_size: Int (default: 32).
+        shuffle: Boolean (default: True).
+        sample_weight: Sample weights.
+        seed: Int (default: None).
+        save_to_dir: None or str (default: None).
+            This allows you to optionally specify a directory
+            to which to save the augmented pictures being generated
+            (useful for visualizing what you are doing).
+        save_prefix: Str (default: `''`).
+            Prefix to use for filenames of saved pictures
+            (only relevant if `save_to_dir` is set).
+        save_format: one of "png", "jpeg"
+            (only relevant if `save_to_dir` is set). Default: "png".
+        subset: Subset of data (`"training"` or `"validation"`) if
+            `validation_split` is set in `ImageDataGenerator`.
+    """
+```
+
+```python
+# PROPOSED
+
+def array_to_generator(
```

Comment:
There seems to be some variation in from/to -- should these be standardized?

Reply:
Yes, let's standardize them. We can either settle for (input)_to_(output) or (output)_from_(input)